redis/asyncio/connection.py

```diff
@@ -650,7 +650,7 @@ def __init__(
             retry_on_error.append(socket.timeout)
             retry_on_error.append(asyncio.TimeoutError)
         self.retry_on_error = retry_on_error
-        if retry_on_error:
+        if retry or retry_on_error:
```

Comment:
Seems duplicative. First there's a check for retry or retry_on_error, but then there's an explicit retry check. WDYT about reworking this block?